Add wifi scan profiles (#12274)
nrfconnect/sdk-nrf: samples/wifi/scan/README.rst

********

The sample demonstrates Wi-Fi scan operations in the 2.4 GHz and 5 GHz bands.

> The sample allows you to perform scan based on below profiles.

Reviewer: ```suggestion
The sample allows you to perform scans based on the following profiles:
```

* Default scan
* Active scan
* Passive scan
* 2.4 GHz Active scan
* 2.4 GHz Passive scan
* 5 GHz Active scan
* 5 GHz Passive scan

> * Scan only non overlapping channels in 2.4 GHz band

Reviewer: ```suggestion
* Scan only non-overlapping channels in 2.4 GHz band
```

> * Scan only non DFS channels in 5 GHz band

Reviewer: ```suggestion
* Scan only non-DFS channels in 5 GHz band
```
